sdk/storage/Azure.Storage.Queues/src/Models/DequeuedMessage.cs

```diff
@@ -15,10 +12,28 @@ namespace Azure.Storage.Queues.Models
     /// </summary>
     public partial class DequeuedMessage
     {
+        /// <summary>
+        /// Cacched message text as bytes.
+        /// </summary>
+        private byte[] _bytes = null;
+
+        /// <summary>
+        /// Gets the <see cref="MessageText"/> as bytes.
+        /// </summary>
+        /// <returns>The <see cref="MessageText"/> as bytes.</returns>
+        public byte[] GetMessageBytes()
```

Comment:
FWIW - This API gives me pause.  I think the problem is that this is a semantics we are laying on top of the service, and I just don't know if folks mental model is going to push them into the right place.  My concern is around the fact that this Base64 stuff just feels like smuggling on top of the service, but we elevate it here to something that feels much more like a first class concept.

Specifically, this method only works when the message was previously encoded in Base64.  You can't call it on an arbitrary message and expect that its going to the right thing.  You have to know that the message was published and encoded as Base64.  Maybe that's the common (the portal, for example, seems to push you towards sending Base64 encoded messages and Azure functions require that the data be Base64 encoded if you set up queue trigger), of course, in that case the message text itself is going to be nonsense and you'd have to base64 decode that (and build a string from it) to do something reasonable.

I sort of think I would feel better if these were extension methods, and in some Azure.Storage.Queues.Binary namespace or something?

I dunno, I just feel a little nervous.  If this is coming from user studies, then I'm happy to be wrong, but if we are just proactively adding these late because we think it will help customers, I think maybe we should just hold off on doing this for the next release till we get feedback from users that they want a convenient way to do this

Reply:
We're doing this specifically for interop scenarios.  Some of the other, existing client libraries do this by default for every message.  You could argue the correctness of that _(and I do...)_, but we still want it to be easy for customers to interop with those libraries.

We could maybe put these in the `.Specialized` namespace as extension methods, but then we hit the problem that they can't be mocked.  Thoughts @KrzysztofCwalina?

Reply:
I would not use extension methods for it. I think we should either ship the method or remove it, and think about the binary scenarios later.

BTW, I share Matt's concerns, but us exposing byte[] could be much more efficient than users doing base64 decoding in  their code. We have access to the response buffer (UTF8) and we can based64 decode UTF8 text directly. If users do the decoding, we have to first transcode from UTF8 to UTF16, and then base64 decode.